opengrok-indexer/src/main/java/org/opengrok/indexer/history/History.java

```diff
@@ -123,4 +123,27 @@ public boolean isRenamed(String file) {
     public Set<String> getRenamedFiles() {
         return renamedFiles;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        History that = (History) o;
+        return this.getHistoryEntries().equals(that.getHistoryEntries()) &&
```

Comment:
I'd advise using the newer comparison as it can handle theoretical NPEs even in the future: https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Objects.html#equals(java.lang.Object,java.lang.Object)

You can let IDEA generate it as well if you choose Java7+ equals/hashcode option.

Reply:
Exactly what I needed ! Thanks !